src/ngMock/angular-mocks.js

```diff
@@ -1683,6 +1683,17 @@ function MockXhr() {
   };
 
   this.abort = angular.noop;
+
+  this.$$events = {};
+  this.addEventListener = function(name, listener) {
+    if (angular.isUndefined(this.$$events[name])) this.$$events[name] = [];
+    this.$$events[name].push(listener);
+  };
+
+  this.upload = {
+    $$events: {},
+    addEventListener: this.addEventListener
+  };
```

Comment:
That makes sense and could be implemented like [MockWindow's fire method](https://github.com/angular/angular.js/blob/master/test/ng/browserSpecs.js#L41-L45) .
However, I don't want to put too many changes in this PR at the moment as it'll affect reviewing, and the chances of it getting merged.